db/config.cc

```diff
@@ -1090,6 +1090,9 @@ db::config::config(std::shared_ptr<db::extensions> exts)
         "the same endpoint used in the request. The string 'disabled' "
         "disables the DescribeEndpoints operation. Any other string is the "
         "fixed value that will be returned by DescribeEndpoints operations.")
+    , alternator_allow_system_table_write(this, "alternator_allow_system_table_write", liveness::LiveUpdate, value_status::Used,
```

Comment:
isn't this going around access rights? I think non super user won't have access to system tables (applies for reading too)

Reply:
We don't have any ACLs implemented in Alternator, so I'm not sure what's the point for enforcing them only on system tables, but maybe we should?
In any case, if CQL has ACLs also for **reading** system tables, then this is a pre-existing problem because we don't check them when reading through Alternator....

Reply:
- it may be preexisting but you are opening the security hole even more because with reading only you still have salted hash while with writing you can set your password
- even if we don't yet have permission checks in alternator we will, and even currently we allow cql with alternator.

I am not against it as it's useful for testing but perhaps clearly mark that those system table reads and writes are only for debugging or testing? i.e. prefix with unsafe word?

Reply:
It's not *just* for debugging and testing - a user might want to allow this if they want to change configuration on the fly using the DynamoDB API, for example, but I agree, it has security implications.

Though I don't think the problem is serious: If you use Alternator, you know there is *zero* ACLs. A user who has valid access to your database can delete all your tables and (with this patch) also change the configuration and even change the password. I made this new feature optional to reduce this one last risk (but not the first ones I listed!) but I don't think there's any way to solve this without implementing some sort of ACL.

Reply:
Since we have (nearly) ACL, this can be re-evaluated/addressed.